### PR TITLE
[Go] Concepts List WIP

### PR DESCRIPTION
**Pull request by tehsphinx**
Here is a list of Concepts I have started in the early discussions around `v3`. This is by no means a finished list!

**I would like to use this PR as a discussion ground. Everything is in question on this list! Not only the changes I made in this PR.**

My approach for this list was looking through the 3 links mentioned at the bottom and see what features/concepts Go has. Another approach would be to look at an exercise and break down which concepts are needed to solve it. Yet another approach would be to look at other tracks (e.g. C#, JavaScript) and see what concepts they use and whether that applies to Go. For example we do not have the concept of `Strings` in here yet where we could teach students about working with strings (e.g. concatenation, manipulation, iteration, etc.). Maybe that even needs multiple concept exercises.

### CODE REVIEW

File: languages/go/reference/README.md [hunk 1]
```diff
@@ -1,7 +1,11 @@
 # Go
 
+This is a list of concepts of Go. They are categorized into `OOP`, `Functional` and `General`. The `Specialties` category tries to mention concepts that are somewhat special to Go. A concept can belong to multiple categories.
+
 ## Specialties
 
+Concepts that are special to Go or are used differently to a vast majority of other languages.
+
```
Comment (sebito91) [inline languages/go/reference/README.md]: Would something like `Code Generation` be a fitting example? This is not specific only to go, but is something pretty advanced within the go world.

Comment (tehsphinx) [reply languages/go/reference/README.md]: Would be a pretty advanced concept. I like it!

**Comment by tehsphinx** [comment]
@sebito91 Is the idea clear of this PR or do you think it is better to remove the file and make a new PR with the whole file being added. That way the entire file is being reviewed and discussed...

What do you think?

**Comment by sebito91** [comment]
Honestly I think this should be an Issue instead of PR until we're ready to fill out the details. That way we can collaborate without submitting multiple reviews.

I was thinking perhaps of having an `Advanced` section in this as well that could house a few more relevant topics.

**Comment by tehsphinx** [comment]
Sounds good. I will create an issue and close this for now.

**Pull request closed without merging**